dora/core/client/fs/src/main/java/alluxio/client/file/dora/ConsistentHashProvider.java

```diff
@@ -66,26 +68,26 @@ public class ConsistentHashProvider {
    * Used to compare with incoming worker list to skip the heavy build process if the worker
    * list has not changed.
    */
-  private final AtomicReference<List<BlockWorkerInfo>> mLastWorkerInfos =
-          new AtomicReference<>(ImmutableList.of());
+  private final AtomicReference<List<WorkerIdentity>> mLastWorkers =
```

Comment:
Question:

The `WorkerIdentity` is used to identify a worker uniquely in the hash ring.
But why is this change needed?
Can we keep the original code, and use the BlockWorkerInfo::WorkerIdentity if the `WorkerIdentity` is needed to compute and maintain the hash ring?

Reply:
Only `WorkerIdentity` is needed to compute the hash ring. Anything else is out of concern of `ConsistentHashProvider`. `BlockWorkerInfo` contains volatile fields such as worker network address and cache usage information, which should not be cached by `ConsistentHashProvider`. This fix is all about removing unnecessary info from the hash ring, and let the callers from higher levels to figure out the details of a selected worker.